### ISSUE TITLE
a11y accessibility plugin slow on larger components

### ISSUE THREAD

**Issue by DerJacques** [storybooks/storybook#3683]
### Bug or support request summary

The a11y accessibility plugin for good reasons takes some time to run its analysis.
We have a somewhat large component and it takes the plugin about 3 seconds to complete the analysis.

From reading the code, it appears that the accessibility analysis is run as soon as the component is mounted. Maybe it would make sense to run the analysis on request only?

### Steps to reproduce
1. Start a performance recording in chrome
2. Mount a somewhat large component
3. Search for "runA11yCheck"

I have attached a screenshot of the local test results.

### Please specify which version of Storybook and optionally any affected addons that you're running

- @storybook/react 3.4.3
- @storybook/addon-a11y 3.4.3

### Affected platforms 
Chrome, Mac OS

### Screenshots / Screencast / Code Snippets (Optional)
![screen shot 2018-05-29 at 16 18 02](https://user-images.githubusercontent.com/3896442/40664887-39b5cf56-635c-11e8-87f2-221b742ac4e2.png)

### Solution suggestion
I think that it would make sense to run the analysis _on demand_ only. So whenever the accessibility tab is actually opened. As of right now, the analysis is always run.

**Comment by ndelangen** [comment]
@DerJacques I see that becoming a problem! I'd be happy to take a PR to change the behavior.

Do you think you'd be up for it? any help you'd need from me?

**Comment by ndelangen** [comment]
@DerJacques I'd be more then happy to help you with that! Would you care to join our slack and we can possibly setup a online meeting where I can assist you.

[invite to slack](https://now-examples-slackin-rrirkqohko.now.sh/)

**Comment by DerJacques** [comment]
@Keraito, thank you for the great explanation! My dev environment works perfectly now 👌

Back to the issue at hand:
My first intuition was to move the following line from (https://github.com/storybooks/storybook/blob/dd867d2ca0d369c43dcd92477c3d3cd663aef243/addons/a11y/src/index.js#L26)[/a11y/src/index.js#L26] to the addon panel's `componentDidMount` hook:

```js
// storybook/addons/a11y/src/components/Panel.js
componentDidMount() {
  this.channel.on(Events.STORY_RENDERED, runA11yCheck);
  this.channel.on(CHECK_EVENT_ID, this.onUpdate);
}
```

However, it seems that the addon panels are actually all mounted and rendered all the time. They are just hidden using a `display: none`.

Is there any other way to check if an addon has been selected? Feels a bit bad to look at the URL for this, but that would be a possibility.

Changing the behaviour from `display: none` to a conditional rendering may be a different way to go, but this would potentially be a breaking change and may have unforeseen consequences.

**Comment by DerJacques** [comment]
Thanks @ndelangen for the invitation! I am a bit busy at the moment but will try to get in touch asap. :)

**Comment by ndelangen** [comment]
@DerJacques I'm working on that!

https://github.com/storybooks/storybook/blob/d0b507ec8bc4aeb1868b5f571002a79d7e8b6409/addons/a11y/src/register.js#L13

**Comment by DerJacques** [comment]
@ndelangen Fantastic! I guess it makes sense to wait for your branch to be merged in? :) If you remember to update this issue when the change is made that would be great. If not, I will make sure to check back in regularly.

**Comment by DerJacques** [comment]
Fixed in https://github.com/storybooks/storybook/pull/3690.
